components/security/src/lib.rs

```diff
@@ -276,10 +307,35 @@ fn check_common_name(
     }
 }
 
-/// Check peer CN with a set of allowed CN.
-pub fn match_peer_names(allowed_cn: &HashSet<String>, name: &str) -> bool {
-    for cn in allowed_cn {
-        if cn == name {
+/// Check peer SAN with cert-allowed-san field.
+/// Similar to check_common_name, but allows for a match on any available certificate SAN (there
+/// may be multiple).
+fn check_subject_alternative_name(
+    cert_allowed_san: &HashSet<String>,
+    ctx: &RpcContext<'_>,
+) -> Result<(), String> {
+    if let Some(auth_ctx) = ctx.auth_context() {
+        let auth_property_sans = auth_ctx
+            .into_iter()
+            .filter(|x| x.name() == "x509_subject_alternative_name");
+
+        for auth_property in auth_property_sans {
+            let peer_san = auth_property.value_str().unwrap();
+            if match_peer_names(cert_allowed_san, peer_san) {
+                return Ok(());
+            }
+        }
+
+        Err("no matching x509_subject_alternative_name or doesn't exist".to_string())
+    } else {
+        Ok(())
+    }
+}
+
+/// Check peer name with a set of allowed CN or SANs.
+pub fn match_peer_names(allowed_names: &HashSet<String>, name: &str) -> bool {
+    for cert_name in allowed_names {
+        if name.contains(cert_name) {
```

Comment:
this would allow some flexibility when specifying the allowed san/cn entry, let me know if it helps, if not I can revert to strict `==`